src/CypherEscrow.sol

```diff
@@ -180,20 +177,22 @@ contract CypherEscrow is ReentrancyGuard {
     function denyTransaction(bytes32 key, address to) external onlyOracle nonReentrant {
         Transaction memory txInfo = getTransactionInfo[key];
 
+        // update storage first to prevent reentrancy
+        delete getTransactionInfo[key];
+
         // Send ETH back
         if (txInfo.asset == address(0x0)) {
             (bool success, ) = address(to).call{value: txInfo.amount}("");
             if (!success) revert TransferFailed();
         } else {
             // Send ERC20 back
             /// TODO: this could be a potential exploit
-            address token = txInfo.asset;
+            address token_ = txInfo.asset;
```

Comment:
Is there a reason for this?

Reply:
it conflicts with a read from state `address public token;`

Reply:
lets name it `asset` then